### ISSUE TITLE
feature request: Gist support

### ISSUE THREAD

**Issue by millermedeiros** [joemccann/dillinger#8]
based on 2 twits from @leaverou:

> Wish I had time for a Markdown version of dabblet: An online md editor that saved to github gists. Can't seem to find anything like it :(
> https://twitter.com/#!/LeaVerou/status/154585603516809218

and

> Dillinger http://dillinger.io/ looks quite interesting, although it's integrated with github repos, not gists.
> https://twitter.com/#!/LeaVerou/status/154587930374049793

**Comment by odedw** [comment]
Hey @joemccann, I'd like to take a look at this. I think it'll be easier if gists were separate from repositories meaning you'll have a "save to github" as well as a "save to gist" option in the menu. Authentication can be different as well as gists are a separate scope, or we can authenticate both togetheer.

What do you think?

**Comment by joemccann** [comment]
Totally open to adding gist support but it needs to be another plugin,
meaning, app code, UI code, interaction design, etc. all needs to be hashed
out.  I'm all for adding it if it adheres to current Dillinger approach

**Comment by odedw** [comment]
I agree that the UI code should be separate, but why should it be a separate plugin? don't you think it'll be easier to add functionality to the existing GitHub plugin? there's probably a lot of code there that we can reuse no?

**Comment by joemccann** [comment]
I suppose we could extend the github plugin code but it may make sense to
unbundle the github plugin itself to have GH-repos an GH-gists or something
along those lines.  Make sense? Gists and repos APIs have their own version
on GH so bundling them under one repo concerns me

**Comment by odedw** [comment]
Makes sense. I'll try to get into it and let you know how it goes.
